internal-functions

```diff
@@ -75,3 +75,49 @@ fn-acl-check-app() {
     dokku_log_fail "You can only modify ACL using local dokku command on target host"
   fi
 }
+
+fn-acl-check-service() {
+  declare SERVICE_TYPE="$1" SERVICE="$2"
+
+  local SERVICE_PATH="$DOKKU_LIB_ROOT/services/$SERVICE_TYPE/$SERVICE"
+  if ! [[ -d $SERVICE_PATH ]]; then
+    dokku_log_fail "Service $SERVICE of type $SERVICE_TYPE does not exist"
+  fi
+
+  if [[ -n "${NAME:-}" ]]; then
+    dokku_log_fail "You can only modify ACL using local dokku command on target host"
+  fi
+}
+
+check-app-acl() {
+  declare desc="Checks if the current user has an ACL entry for the app"
+  declare APP=$1
+
+  if ! ( verify_app_name "$APP" 2>/dev/null ); then
```

Comment:
You don't need to wrap this in parenthesis.

Reply:
The parentheses are needed in order to run `verify_app_name` in a subshell. Without this, if the app doesn't exist it calls `dokku_log_fail` and the hook exits with "App $APP does not exist" rather than allowing dokku-acl to print its own message.